internal/services/sources_service.go

```diff
@@ -19,11 +20,19 @@ func ListSources(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	resp, err := client.ListApplicationTypeSourcesWithResponse(r.Context(), config.Sources.AppTypeId, &sources.ListApplicationTypeSourcesParams{}, AddIdentityHeader)
+	statusCode := resp.StatusCode()
 	if err != nil {
-		renderError(w, r, payloads.New3rdPartyClientError(r.Context(), "list sources", err))
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "list sources", err, statusCode))
+		return
+	}
+	if parsing.IsHTTPNotFound(statusCode) {
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "sources not found", err, statusCode))
+		return
+	}
+	if !parsing.IsHTTPStatus2xx(statusCode) {
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "sources client error", err, statusCode))
```

Comment:
Wodering if this is a problem, when running the test the status received is 0 which is indeed not a 2xx status code, so the operation resulted in that error.
```golang
 	handler.ServeHTTP(rr, req) // the error is received here which causes the test to fail
	if status := rr.Code; status != http.StatusOK {
		t.Errorf("Handler returned wrong status code. Expected: %d. Got: %d.", http.StatusOK, status)
	}
```

Reply:
here @ezr-ondrej

Reply:
can we change the test to return 200 status?

Reply:
I didn't test, but this IMHO fixes the issue: https://gist.github.com/ezr-ondrej/a5d6d8e5da3525766bc9e6bc22c51c59